executor/runtime/types/container.go

```diff
@@ -1224,7 +1224,7 @@ func populateContainerEnv(c Container, config config.Config, userEnv map[string]
 		env["NETFLIX_NETWORK_MODE"] = netMode
 	}
 	vpcAllocation := c.VPCAllocation()
-	if a := vpcAllocation.IPV4Address(); a != nil {
+	if a := vpcAllocation.IPV4Address(); a != nil && c.EffectiveNetworkMode() != titus.NetworkConfiguration_Ipv6AndIpv4Fallback.String() {
```

Comment:
If we're in IPv6-only mode, I'd set the `EC2IPv4EnvVarName` to `127.0.0.1` so that software which requires a valid value there does not break in an undesired way.